errbot/core.py

```diff
@@ -617,6 +619,16 @@ def callback_stream(self, stream):
         log.info('Initiated an incoming transfer %s.', stream)
         Tee(stream, self.plugin_manager.get_all_active_plugins()).start()
 
+    def callback_reaction(self, reaction):
```

Comment:
Are there concerns on how this would work with other backends?

Reply:
> Are there concerns on how this would work with other backends?

Thats a good question, honestly I hadn't attempted to test with other non-default backends enabled.  

I'd expect that it wouldn't be a significant issue because the other messaging backends  would never trigger the method themselves.   However in multiple messaging backend environments (e.g Slack and XMPP enabled), Slack's backend would trigger callback_reaction which in turn would dispatch the event to all plugins, which would result in caught AttributeError exceptions for those plugins/backends that don't support reactions, leaving a log message to that effect.  

If there is a way to identify pass the event to only Slack plugins, that would be optimal.  I tried to follow the current convention for consistency and simplicity, rather than trying to figure out which plugins should receive the event.  I'm not even certain if that's possible to be honest.

Reply:
> > Are there concerns on how this would work with other backends?
> 
> Thats a good question, honestly I hadn't attempted to test with other non-default backends enabled.
> 
> I'd expect that it wouldn't be a significant issue because the other messaging backends would never trigger the method themselves. However, Slack's backend could trigger callback_reaction which in turn would dispatch the event to all plugins for all other backends. That would result in caught AttributeError exceptions for those plugins/backends that don't support reactions, leaving a log message to that effect.
> 

So that makes sense how that would work. 

> If there is a way to identify pass the event to only Slack plugins, that would be optimal. I tried to follow the current convention for consistency and simplicity, rather than trying to figure out which Slack specific plugins should receive the event. I'm not even certain if that's possible to be honest.

Backends do have a `mode` [property](https://github.com/errbotio/errbot/blob/master/errbot/backends/slack.py#L960), that maybe can be used.